org.eclipse.acute.SWTBotTests/src/org/eclipse/acute/SWTBotTests/dotnetrun/TestRun.java

```diff
@@ -0,0 +1,84 @@
+package org.eclipse.acute.SWTBotTests.dotnetrun;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.swt.widgets.Tree;
+import org.eclipse.swtbot.eclipse.finder.widgets.SWTBotEditor;
+import org.eclipse.swtbot.eclipse.finder.widgets.SWTBotView;
+import org.eclipse.swtbot.swt.finder.finders.ChildrenControlFinder;
+import org.eclipse.swtbot.swt.finder.matchers.WidgetOfType;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotStyledText;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotTree;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotTreeItem;
+import org.junit.Test;
+
+public class TestRun extends AbstractRunTest{
+	protected IFile csharpSourceFile;
+	private String expectedOutput = "Hello World!\n";
+	
+	@Override
+	public void setup() throws CoreException {
+		super.setup();
+		buildDotnetProject();
+		
+		SWTBotView view = bot.viewByTitle("Project Explorer");
+		List<Tree> controls = new ChildrenControlFinder(view.getWidget()).findControls(WidgetOfType.widgetOfType(Tree.class));
+		SWTBotTree tree = new SWTBotTree((Tree) controls.get(0));
+		SWTBotTreeItem projectItem = tree.getTreeItem(project.getName());
+		SWTBotTreeItem fileItem = projectItem.expand().getNode(csharpSourceFile.getName());
+		fileItem.select().contextMenu("Open").click();
+		SWTBotEditor editor = bot.editorByTitle("Project.cs");
+		editor.setFocus();
+	}
+
+	@Test
+	public void testDotnetRun() throws CoreException, InterruptedException {
+		bot.styledText(1).contextMenu("Run As").menu("1 .NET Project").click();
+
+		SWTBotView consoleView = bot.viewByPartName("Console");
+		consoleView.setFocus();
+		
+		SWTBotStyledText consoleText = consoleView.bot().styledText();
+		int delay = 60;
+		while(consoleText.getText().isEmpty() && delay > 0) {
+			Thread.sleep(500);
+			consoleText = consoleView.bot().styledText();
+			delay--;
+		}
+		if(delay == 0) {
+			fail("Run as function timed out after 30 seconds.");
+		}
+		String actualOutput = consoleText.getText();
+		assertEquals("Expected output does not match actual.", expectedOutput, actualOutput);
+	}
+	
+	public void buildDotnetProject() {
+		try {
+			csharpSourceFile = this.project.getFile("Project.cs");
+			csharpSourceFile.create(getClass().getResourceAsStream(csharpSourceFile.getName()), true, new NullProgressMonitor());
+			
+			IFile csproj = this.project.getFile("project.csproj");
+			csproj.create(getClass().getResourceAsStream(csproj.getName()), true, new NullProgressMonitor());
+			
+			ProcessBuilder processBuilder;
+			processBuilder = new ProcessBuilder("dotnet", "restore");
```

Comment:
I don't think this should be part of tests. I would expect the `Run As` to automatically call `dotnet restore` if necessary.